src/main/ts/components/useData.ts

```diff
@@ -65,41 +67,47 @@ export default function useData(): UseDataReturn {
 
   useEffect(() => {
     (async () => {
-      const issueRepartitionData = await getIssues();
+      try {
+        const issueRepartitionData = await getIssues();
 
-      let cumulativeIssueCount = 0;
-      const cumulativeData = issueRepartitionData.map(({ x, y }) => {
-        cumulativeIssueCount += y;
-        return {
-          x,
-          y: cumulativeIssueCount,
-        };
-      });
+        setHasRequestFailed(false);
 
-      setCumulativeData(cumulativeData);
+        let cumulativeIssueCount = 0;
+        const cumulativeData = issueRepartitionData.map(({ x, y }) => {
+          cumulativeIssueCount += y;
+          return {
+            x,
+            y: cumulativeIssueCount,
+          };
+        });
 
-      // x scale
-      const dateRange = extent(cumulativeData, (d) => d.x) as [Date, Date];
-      const timeScale = scaleTime().range([0, GRAPH_WIDTH]).domain(dateRange).clamp(false);
-      setXScale(() => timeScale);
-      setChartDateRange(dateRange);
+        setCumulativeData(cumulativeData);
 
-      // y scale
-      const linearScale = scaleLinear()
-        .range([GRAPH_HEIGHT - 16, 0])
-        .domain([0, cumulativeIssueCount * 1.5])
-        .nice();
-      setYScale(() => linearScale);
+        // x scale
+        const dateRange = extent(cumulativeData, (d) => d.x) as [Date, Date];
+        const timeScale = scaleTime().range([0, GRAPH_WIDTH]).domain(dateRange).clamp(false);
+        setXScale(() => timeScale);
+        setChartDateRange(dateRange);
 
-      // cayc available periods
-      const [chartStartDate, chartEndDate] = dateRange;
-      const caycAvailablePeriods = CAYC_DURATIONS.filter(
-        (duration) => chartStartDate < sub(chartEndDate, duration)
-      ).map((duration, i) => ({ duration, value: i }));
-      setCaycAvailableDurations(caycAvailablePeriods);
-      setCurrentCaycDuration(caycAvailablePeriods[caycAvailablePeriods.length - 1]);
+        // y scale
+        const linearScale = scaleLinear()
+          .range([GRAPH_HEIGHT - 16, 0])
+          .domain([0, cumulativeIssueCount * 1.5])
+          .nice();
+        setYScale(() => linearScale);
 
-      setLoading(false);
+        // cayc available periods
+        const [chartStartDate, chartEndDate] = dateRange;
+        const caycAvailablePeriods = CAYC_DURATIONS.filter(
+          (duration) => chartStartDate < sub(chartEndDate, duration)
+        ).map((duration, i) => ({ duration, value: i }));
+        setCaycAvailableDurations(caycAvailablePeriods);
+        setCurrentCaycDuration(caycAvailablePeriods[caycAvailablePeriods.length - 1]);
+      } catch (e) {
```

Comment:
This try/catch block contains too many stuffs I believe. It's probably better to narrow it down to just the API call.

Reply:
Yeah, you're right